File: code-test/solutions/java/Day011.java
import java.util.*;

public class Day011 {
        static List<List<Integer>> graph;
        static boolean[] visited;

    public static int solution(int n, int[][] edges) {
        // TODO: 무방향 그래프의 연결 요소 개수를 반환하세요.
        // 예: n = 5, edges = [[1, 2], [2, 3], [4, 5]] -> 2
        graph = new ArrayList<>();
        visited = new boolean[n + 1];
        int answer = 0;

        for(int i=0; i<=n; i++) {
            graph.add(new ArrayList<>());
        }

        for(int[] edge: edges) {
            int a = edge[0];
            int b = edge[1];
        
            graph.get(a).add(b);
            graph.get(b).add(a);
        }

        for (int i=1; i<= n; i++) {
            if (!visited[i]) {
                answer += 1;
                dfs(i);
            }
        }



        return answer;
    }
    
    static void dfs(int node) {
        visited[node] = true;

        for (int next : graph.get(node)) {
            if (! visited[next]) {
                dfs(next);
            }
        }
    }
}

File: code-test/solutions/python/day_011.py
def solution(n, edges):
    # TODO: 무방향 그래프의 연결 요소 개수를 반환하세요.
    # 예: n = 5, edges = [[1, 2], [2, 3], [4, 5]] -> 2
    graph = [[] for _ in range(n + 1)]

    for a, b in edges:
        graph[a].append(b)
        graph[b].append(a)

    visited = [False] * (n + 1)
    answer = 0

    def dfs(node):
        visited[node] = True

        for next in graph[node]:
            if not visited[next]:
                dfs(next)

    for node in range(1, n + 1):
        if not visited[node]:
            answer += 1
            dfs(node)

    return answer
